[CI] EsqlSpecIT test {categorize.On REVERSE(CONCAT()) ASYNC} failing

**Build Scans:**
- [elasticsearch-periodic #8851 / openjdk17_checkpart3_java-fips-matrix](https://gradle-enterprise.elastic.co/s/cbfs7faciapdi)
- [elasticsearch-periodic #8851 / openjdk17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/wbgiz6sr3mu3o)
- [elasticsearch-periodic #8851 / graalvm-ce17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/qon7mhabdq2vc)
- [elasticsearch-periodic #8845 / openjdk17_checkpart3_java-fips-matrix](https://gradle-enterprise.elastic.co/s/b6w3upt3vi5ji)
- [elasticsearch-periodic #8845 / openjdk17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/sbogsfctdlby6)
- [elasticsearch-periodic #8845 / graalvm-ce17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/gygu4lsist3ug)
- [elasticsearch-periodic #8839 / graalvm-ce17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/diyle5itvsyvy)
- [elasticsearch-periodic #8839 / openjdk17_checkpart3_java-fips-matrix](https://gradle-enterprise.elastic.co/s/5hzv2exsh2554)
- [elasticsearch-periodic #8839 / openjdk17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/vhunpufmpkpik)
- [elasticsearch-periodic-platform-support #8842 / ubuntu-2404-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/d4tyn6keeucgm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT" -Dtests.method="test {categorize.On REVERSE(CONCAT()) ASYNC}" -Dtests.seed=37E0D2ABC15B720B -Dtests.locale=bem -Dtests.timezone=Australia/Tasmania -Druntime.java=17 -Dtests.fips.enabled=true
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bcategorize.On%20REVERSE(CONCAT())%20ASYNC%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Data mismatch:
row 0 column 1:a list containing
row 0 column 1:0: expected ".*?😊.+?👍🏽.+?detcennocsiD.*?" but was ".*?😊🏽.+?👍.+?detcennocsiD.*?"
row 1 column 1:a list containing
row 1 column 1:0: expected ".*?😊.+?👍🏽.+?ot.+?detcennoC.*?" but was ".*?😊🏽.+?👍.+?ot.+?detcennoC.*?"
row 2 column 1:a list containing
row 2 column 1:0: expected ".*?😊.+?👍🏽.+?rorre.+?noitcennoC.*?" but was ".*?😊🏽.+?👍.+?rorre.+?noitcennoC.*?"
Actual:
COUNT():long | category:keyword
1            | .*?😊🏽.+?👍.+?detcennocsiD.*?
3            | .*?😊🏽.+?👍.+?ot.+?detcennoC.*?
3            | .*?😊🏽.+?👍.+?rorre.+?noitcennoC.*?

Expected:
COUNT():long | category:keyword
1            | .*?😊.+?👍🏽.+?detcennocsiD.*?
3            | .*?😊.+?👍🏽.+?ot.+?detcennoC.*?
3            | .*?😊.+?👍🏽.+?rorre.+?noitcennoC.*?

```

**Issue Reasons:**
- [8.19] 3 consecutive failures in step openjdk17_checkpart3_java-fips-matrix
- [8.19] 3 consecutive failures in step openjdk17_checkpart3_java-matrix
- [8.19] 3 consecutive failures in step graalvm-ce17_checkpart3_java-matrix
- [8.19] 10 failures in test test {categorize.On REVERSE(CONCAT()) ASYNC} (6.5% fail rate in 153 executions)
- [8.19] 3 failures in step openjdk17_checkpart3_java-fips-matrix (60.0% fail rate in 5 executions)
- [8.19] 3 failures in step openjdk17_checkpart3_java-matrix (75.0% fail rate in 4 executions)
- [8.19] 3 failures in step graalvm-ce17_checkpart3_java-matrix (75.0% fail rate in 4 executions)
- [8.19] 3 failures in pipeline elasticsearch-periodic (60.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/categorize.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/categorize.csv-spec
@@ -309,6 +309,7 @@ COUNT():long | category:keyword
 
 on REVERSE(CONCAT())
 required_capability: categorize_v6
+required_capability: fn_reverse_grapheme_clusters
 
 FROM sample_data
   | STATS COUNT() BY category=CATEGORIZE(REVERSE(CONCAT(message, " 👍🏽😊")))

